src/leiningen/monolith.clj

```diff
@@ -274,28 +275,68 @@
     (fingerprint/clear project opts more)))
 
 
+(defn ^:higher-order with-dependency-set
+  "Run a task with a set of managed dependencies from a named dependency set.
+
+   Overrides the dependencies from the named dependency set into the project.
+   For the root project, this means the managed dependencies will be overwritten
+   with the dependencies from the named set. For subprojects, the
+   `:monolith/dependency-set` metadata key will be set to the named set. If the
+   project does not have the `:monolith/dependency-set` key defined, it will be
+   added.
+
+   For combining with other higher-order tasks, this task should be the last
+   evaluated.
+
+   A gotcha: this task does not update the project's `project.clj` file, so the
+   dependencies will not be saved to disk. This is intentional, as the task is
+   meant to be used for temporary operations that need to be run with a specific
+   set of dependencies. This also means that the task will not work with tasks
+   that require the project file to be updated.
```

Comment:
Do we need this comment? As a user, my default assumption is that build tool commands aren't going to modify my source code files unless they explicitly say they will. Also, (ephemeral) project modification _can_ be accomplished using leiningen's built in `update-in` task - though I don't think we need that in the actual docstring.

I'm inclined towards somewhat more brevity here since all this would be output by `lein help monolith with-dependency-set`. If we want to have more details docs, perhaps adding some task-specific information under `doc/tasks.md` would be useful?

Reply:
This makes sense to me, I'll make the change